automation-generators/aws/openshift/main.yaml

```diff
@@ -1 +1,65 @@
 ---
+- name: "Start {{generator.name}} generator"
+  debug:
+    msg: 'from {{generator.name}} in {{generator.path}} attributes: {{generator.attributes}}'
+
+- name: Show generator
+  debug:
+    var: generator
+
+- set_fact:
+    _openshift_config: "{{ all_config.openshift | json_query(query) | first | default({}) }}"
+  vars:
+    query: >-
+      [?name=='{{ generator.attributes.name }}']
+
+- name: Get aws-access-key secret from in the vault
+  include_role: 
+    name: vault-get-secret
+  vars:
+    secret_name: "aws-access-key"
+    secret_group: "{{ environment_name }}" 
+
```

Comment:
You can directly populate the secret variable by adding the _p_secret_variable argument. Example:
```
- name: Get aws-access-key secret from in the vault
  include_role: 
    name: vault-get-secret
  vars:
    secret_name: "aws-access-key"
    secret_group: "{{ environment_name }}" 
    _p_secret_variable: _aws_access_key
```